agent/consul/client.go

```diff
@@ -89,58 +88,30 @@ type Client struct {
 	tlsConfigurator *tlsutil.Configurator
 }
 
-// NewClient is used to construct a new Consul client from the configuration,
-// potentially returning an error.
-// NewClient only used to help setting up a client for testing. Normal code
-// exercises NewClientLogger.
-func NewClient(config *Config) (*Client, error) {
-	c, err := tlsutil.NewConfigurator(config.ToTLSUtilConfig(), nil)
-	if err != nil {
-		return nil, err
-	}
-	return NewClientLogger(config, nil, c)
-}
-
 func NewClientWithOptions(config *Config, options ...ConsulOption) (*Client, error) {
```

Comment:
Since your removing NewClient should you also rename this to NewClient to be more succinct?

Reply:
Good call on these renames! I'll make that change